tests/end_to_end/helpers.h

```diff
@@ -307,6 +307,52 @@ typedef class _test_xdp_helper
     }
 } test_xdp_helper_t;
 
+// These are test xdp context creation functions.
+_Success_(return == 0) static uint64_t _xdp_context_create(
+    _In_reads_bytes_opt_(data_size_in) const uint8_t* data_in,
+    _In_ size_t data_size_in,
+    _In_reads_bytes_opt_(context_size_in) const uint8_t* context_in,
+    _In_ size_t context_size_in,
+    _Outptr_ void** context)
+{
+    if (context_in != NULL || context_size_in != 0) {
+        return 1;
+    }
+
+    xdp_md_t* xdp_context = reinterpret_cast<xdp_md_t*>(malloc(sizeof(xdp_md_t)));
+    if (xdp_context == nullptr) {
+        return 1;
+    }
+
+    xdp_context->data = (void*)data_in;
+    xdp_context->data_end = (void*)(data_in + data_size_in);
+    *context = xdp_context;
```

Comment:
what about initializing xdp_context->data_meta and xdp_context->ingress_ifindex?

Reply:
it seems odd to silently ignore (and not fail) the context passed in if 0 < context_size_in < sizeof(xdp_md_t).  Seems like it should either fail, or else copy whatever portion was provided and zero pad the rest.